### ISSUE TITLE
Status-code must be 3-digit

### ISSUE THREAD

**Issue by zuevmaxim** [ktorio/ktor#1882]
**Ktor Version and Engine Used (client or server and name)**
io.ktor:ktor-server-cio:1.3.2

**Describe the bug**
Status-code must be 3-digit(see https://tools.ietf.org/html/rfc2616#section-6.1.1)

**To Reproduce**
Run code:
```
import io.ktor.http.cio.parseResponse
import io.ktor.utils.io.ByteReadChannel
import kotlinx.coroutines.runBlocking


fun main() {
    listOf(
        """
        HTTP/1.1 0 OK


        """.trimIndent(),
        """
        HTTP/1.1 10 OK


        """.trimIndent(),
        """
        HTTP/1.1 1000 OK


        """.trimIndent(),
        """
        HTTP/1.1 10000 OK


        """.trimIndent()
    ).forEach { test(it) }

}

fun test(http: String) {
    val response = runBlocking { parseResponse(ByteReadChannel(http)) }!!
    println("status=${response.status};statusText={${response.statusText}};headers=[${response.headers}];version=${response.version}")
}

```

**Expected behavior**
An error expected - exception or null result.

**Comment by oleg-larshin** [comment]
Please check the <a href="https://youtrack.jetbrains.com/issue/KTOR-370">following ticket</a> on YouTrack for follow-ups to this issue. GitHub issues will be closed in the coming weeks.

**Comment by vladf2n** [comment]
Hey, I'll work on this!

**Comment by e5l** [comment]
Fixed in 1.5.1
